### PR TITLE
Correct segment counting in SpikeAmpltiudes and SpikeLocations get_data

### LINKED ISSUE #2975: Spike amplitude data for multi-segments is incorrect
When you `get_data` by unit for amplitudes, the sampling is incorrect for multi-segment recordings.

Consider the following code:
```
import spikeinterface.full as si

recording, sorting = si.generate_ground_truth_recording(durations=[10,10], seed=1205)

analyzer = si.create_sorting_analyzer(recording=recording, sorting=sorting)
analyzer.compute(['random_spikes', 'templates', 'spike_amplitudes'])

si.plot_amplitudes(analyzer, segment_index=0)
si.plot_amplitudes(analyzer, segment_index=1)
```
The output for the first plot is
Screenshot 2024-06-05 at 10 29 02
This looks good: each unit has amplitudes based around a common mean.

The second plot looks like this
Screenshot 2024-06-05 at 10 29 07
This doesn't look good. The amplitudes for each unit are scattered randomly.

Issue is in `_get_data` function for `ComputeSpikeAmplitudes`. I'm working on a fix.

### PR DESCRIPTION
**Pull request by chrishalcrow**
Fixes #2975, without losing speed.

The `all_amplitudes` and `all_locations` vectors are flattened, so don't take account of segment index. This PR adds a cumulative sum which keeps track of where each segments begins.

In draft until tests are added.

### CODE REVIEW

**Comment by chrishalcrow** [comment]
Note: need to update tests to use fixtures after #2984 is merged.

File: src/spikeinterface/postprocessing/spike_amplitudes.py [hunk 1]
```diff
@@ -127,13 +127,20 @@ def _get_data(self, outputs="numpy"):
         elif outputs == "by_unit":
             unit_ids = self.sorting_analyzer.unit_ids
             spike_vector = self.sorting_analyzer.sorting.to_spike_vector(concatenated=False)
+            segment_cum_count = np.insert(
+                np.cumsum([len(spike_vector[a]) for a in range(0, self.sorting_analyzer.sorting.get_num_segments())]),
+                0,
+                0,
+            )
```
Comment (alejoe91) [inline src/spikeinterface/postprocessing/spike_amplitudes.py:134]: @chrishalcrow I think this should be handled in the `spike_vector_to_indices`, so it's used by all extensions that do this `by_unit`

Comment (alejoe91) [reply src/spikeinterface/postprocessing/spike_amplitudes.py:134]: @samuelgarcia will fix it because it's his fault!

**Comment by samuelgarcia** [comment]
Hi Chris.
I did another proposal instead of this fix see #3048.
The  spike_vector_to_indices() function was ambiguous with relative or absolut. Now I added an option.
We need the two cases depending on the situation.

**Comment by alejoe91** [comment]
This will be replaced by #3048 

@chrishalcrow can we close?

**Comment by chrishalcrow** [comment]
Yes, closing now

**Pull request closed without merging**